### DISCUSSION TITLE
Module `org.junit.jupiter.api` not found, even when specified in `ivyDeps`

### DISCUSSION THREAD

**Discussion by ayhon** [com-lihaoyi/mill#4020] [Q&A]
I have a mill project with two modules, one written in Java and another one written in Scala, where the Scala module depends on the Java one. My team is in charge of the Scala and we're using `mill` as our build system. The Java team instead seems to be using Ecilpse to handle the builds of their project.

I wanted to also include their code inside our `build.sc` file so that I could assemble our code with their code into a Jar file which we could then upload to a server. Recently, they added JUnit5 to handle their tests, and they did so as a runtime dependency, not only for development. Arguably our module shouldn't need to care about what testing framework the other team is using, but in the interest of speed I decided to include it in the `build.sc` anyways. Nevertheless, I was surprised what I found myself unable to do so.

Their module is called `language`. Whenever I run `mill language.compile` I get the following error message
```
[info] compiling 1 Java source to /home/ayhon/uni/soft-proj/projet-gl/out/language/compile.dest/classes ...
[error] /home/ayhon/uni/soft-proj/projet-gl/language/src/module-info.java:9:1: module not found: org.junit.jupiter.api
[error] org.junit.jupiter.api
1 targets failed
language.compile javac returned non-zero exit code
```

The `build.sc` which I'm using is the following:
```scala
import mill._, scalalib._
import $ivy.`com.lihaoyi::mill-contrib-scalapblib:`
import contrib.scalapblib._

val ScalaVersion = "3.5.2"
val ScalaPBVersion = "0.11.17"

object server extends ScalaPBModule {
  // ...  
  def moduleDeps = Seq(language)
  // ...
}

object language extends JavaModule {
  def ivyDeps = Agg(
    ivy"org.junit.jupiter:junit-jupiter-api:5.11.3"
  )
}

```

From what I've been able to tell, `org.junit.jupiter.api` corresponds to https://mvnrepository.com/artifact/org.junit.jupiter/junit-jupiter-api. Nevertheless, `mill` seems to complain that it has not been specified. Has this something to do with how `mill` interacts with `module-info.java`?

<details> 
<summary>
Image of the commands
</summary>

![image](https://github.com/user-attachments/assets/97b2d5e8-b071-44b0-8707-10b19cb46d5c)

</details>

**Comment by lihaoyi** [comment]
I don't think Mill treats `module-info.java` files specially. Would it be possible for you to minimize a reproduction I could look at? At a glance I don't see anything obviously wrong with what you are doing.
